.github/workflows/bazel.yml

```diff
@@ -67,14 +67,154 @@ jobs:
+        with:
+          image: ${{ env.BAZEL_BASE_IMAGE }}
+          shell: bash
+          # TODO: Remove work-around mount of Github workspace to /magma (https://github.com/addnab/docker-run-action/issues/11)
+          options: -v ${{ github.workspace }}:/workspaces/magma/ -v ${{ github.workspace }}/lte/gateway/configs:/etc/magma
+          run: |
+            cd /workspaces/magma
+            set -euo pipefail
+
+            # Required for bazel-diff to perform git checkout.
+            git config --global --add safe.directory /workspaces/magma
+
+            printf '\r%s\r' '###############################' 1>&2
+            printf '\r%s\r' 'Configuring bazel remote cache.' 1>&2
+            printf '\r%s\r\r' '###############################' 1>&2
+            bazel/scripts/remote_cache_bazelrc_setup.sh "${{ env.CACHE_KEY }}" "${{ env.REMOTE_DOWNLOAD_OPTIMIZATION }}" "${{ secrets.BAZEL_REMOTE_PASSWORD }}"
+
+            printf '\r%s\r' '###############################' 1>&2
+            printf '\r%s\r' 'Bazel-diff.' 1>&2
+            printf '\r%s\r' '###############################' 1>&2
+            # Check if the workflow is run on a PR and whether the .bazelrc and WORKSPACE files have been affected.
+            IMPACTED_TARGETS_FILE="/tmp/impacted_targets.txt"
+            if [[ ("${{ github.event_name }}" == "pull_request") && \
+                  ! $(git diff-tree --no-commit-id --name-only -r "${{ github.event.pull_request.base.sha }}" "${{ github.SHA }}" | \
+                    grep -E '(.bazelrc|WORKSPACE.bazel)') \
+                ]];
+            then
+              printf '\r%s\r' 'Running bazel-diff...' 1>&2
+              bazel/scripts/bazel_diff.sh "${{ github.event.pull_request.base.sha }}"  "${{ github.SHA }}" | tee "$IMPACTED_TARGETS_FILE"
+              printf "bazel_diff_ran=true" > /workspaces/magma/bazel_diff_ran.txt
+            else
+              printf '\r%s\r' 'WORKSPACE or .bazelrc files have been modified or trigger is not a pull-request.' 1>&2
+              printf '\r%s\r' 'Skipping bazel-diff ...' 1>&2
+              printf "bazel_diff_ran=false" > /workspaces/magma/bazel_diff_ran.txt
+              exit 0
+            fi
+
+            # Check if impacted_targets.txt is empty, if yes skip all jobs.
+            if [[ ! -s "$IMPACTED_TARGETS_FILE" ]];
+            then
+              printf '\r%s\r' 'No relevant bazel targets are impacted, skipping all jobs.' 1>&2
+              printf "run_package_job=false" > /workspaces/magma/run_package_job.txt
+              printf "" > /workspaces/magma/all_unit_test_targets.txt
+              printf "" > /workspaces/magma/cc_unit_test_targets.txt
+              printf "" > /workspaces/magma/py_service_targets.txt
+              exit 0
+            fi
+
+            # Determine if the release is contained in the impacted targets list.
+            if [[ -z "$(grep '//lte/gateway/release:release_build' "$IMPACTED_TARGETS_FILE")" ]];
```

Comment:
This does not work in bash I think because 
```
if [[ 0 ]];
then
    echo 'This is true'
else
    echo 'This is false'
fi

if [[ 1 ]];
then
    echo 'This is true'
else
    echo 'This is false'
fi
``` 
returns 
```
This is true
This is true
```
You then need to wrap it as a subshell in a string and compare to `0` or so. But that is essentially what it is doing now.

Reply:
I did not wrap it inside [[ ]]
I'm just executing the command, then the if will follow the return code. The ! negates the effect.

Reply:
You can change the line to what I wrote exactly.

Reply:
You are right, omitting the `[[ ]]` is key here. Pushed. I used the long option `--quiet` for better readability.